### PR TITLE
[feat]write tensorboard log for model parameters

### PR DESCRIPTION
**Pull request by eric-gecheng**

### CODE REVIEW

File: tzrec/main.py [hunk 1]
```diff
@@ -336,6 +338,20 @@ def _log_train(
         summary_writer.add_scalar("loss/total", total_loss, step)
         for i, g in enumerate(param_groups):
             summary_writer.add_scalar(f"lr/g{i}", g["lr"], step)
+        for name, param in params.items():
```
Comment (tiankongdeguiji) [inline tzrec/main.py]: add a  summary of norm of gradient and gradient global norm like
https://github.com/alibaba/EasyRec/blob/master/easy_rec/python/compat/optimizers.py#L84 ?

File: tzrec/main.py [hunk 2]
```diff
@@ -428,6 +444,7 @@ def _train_and_evaluate(
                     _log_train(
                         i_step,
                         losses,
+                        params=optimizer.params,
```
Comment (tiankongdeguiji) [inline tzrec/main.py]: add a config for train_config.tensorboard_summaries to control summary what. default is ["learning_rate", "loss"], The summary of histogram might be slow.

**Review by tiankongdeguiji — CHANGES_REQUESTED** [review]

File: tzrec/protos/train.proto [hunk 3]
```diff
@@ -26,5 +26,9 @@ message TrainConfig {
     optional bool use_tensorboard = 10 [default = true];
     // epoch interval for saving checkpoint
     optional uint32 save_checkpoints_epochs = 11;
+    // the summaries to be saved in tensorboard, activated only when use_tensorboard=true,
+    // possible values are: "loss", "learning_rate", "parameter", "global_gradient_norm", "gradient_norm", "gradient"
+    // default values are ["loss", "learning_rate"]
+    repeated string tensorboard_summaries = 12;
```
Comment (tiankongdeguiji) [inline tzrec/protos/train.proto:32]: add the usage to usage/train.md

File: tzrec/main.py [hunk 4]
```diff
@@ -333,25 +334,53 @@ def _log_train(
...
+            for k, v in losses.items():
+                summary_writer.add_scalar(f"loss/{k}", v, step)
+            summary_writer.add_scalar("loss/total", total_loss, step)
+        if "learning_rate" in tb_summaries:
+            for i, g in enumerate(param_groups):
+                summary_writer.add_scalar(f"lr/g{i}", g["lr"], step)
+
+        if "global_gradient_norm" in tb_summaries:
+            global_grad_norm = torch.nn.utils.get_total_norm(
+                [p.grad for p in params.values() if not isinstance(p, ShardedTensor)]
```
Comment (tiankongdeguiji) [inline tzrec/main.py:347]: ShardedTensor does not have gradient?

Comment (eric-gecheng) [reply tzrec/main.py:347]: done

**Review by tiankongdeguiji — APPROVED** [review]
